qabelbox/src/main/java/de/qabel/qabelbox/services/HttpDropConnector.java

```diff
@@ -0,0 +1,191 @@
+                                final Identity identity,
+                                @Nullable final LocalQabelService.OnSendDropMessageResult dropResultCallback)
+            throws QblDropPayloadSizeException {
+        new Thread(new Runnable() {
+            final BinaryDropMessageV0 binaryMessage = new BinaryDropMessageV0(dropMessage);
+            final byte[] messageByteArray = binaryMessage.assembleMessageFor(recipient, identity);
+            HashMap<DropURL, Boolean> deliveryStatus = new HashMap<>();
+
+            @Override
+            public void run() {
+                if (recipient.getDropUrls().size() == 0) {
+                    Log.e(TAG, "no dropurls in recipient");
+                }
+                for (DropURL dropURL : recipient.getDropUrls()) {
+                    HTTPResult<?> dropResult = dropHTTPsend(dropURL, messageByteArray);
+                    if (dropResult.getResponseCode() == 200) {
+                        deliveryStatus.put(dropURL, true);
+                    } else {
+                        deliveryStatus.put(dropURL, false);
+                    }
+                }
+                if (dropResultCallback != null) {
+                    dropResultCallback.onSendDropResult(deliveryStatus);
+                }
+            }
+        }).start();
+    }
+
+    /**
+     * Send DropMessages via DropHTTP. Method extracted to mock send in LocalQabelServiceTester.
+     *
+     * @param dropURL DropURL to send DropMessage to.
+     * @param message Encrypted DropMessage
+     * @return
+     */
+    HTTPResult<?> dropHTTPsend(DropURL dropURL, byte[] message) {
+        return dropHTTP.send(dropURL.getUri(), message);
+    }
+
+
+    /**
+     * Retrieves all DropMessages for given Identities
+     *
+     * @return Retrieved, decrypted DropMessages.
+     */
+    @Override
+    public Collection<DropMessage> retrieveDropMessages(Identity identity, long sinceDate) {
+        Collection<DropMessage> allMessages = new ArrayList<>();
+
+        for (DropURL dropUrl : identity.getDropUrls()) {
+            Collection<DropMessage> results = this.retrieveDropMessages(dropUrl.getUri(), sinceDate);
+            allMessages.addAll(results);
+
+        }
+        return allMessages;
+    }
+
+    /**
+     * Retrieves all DropMessages from given URI
+     *
+     * @param uri URI where to retrieve the drop from
+     * @return Retrieved, decrypted DropMessages.
+     */
+    public Collection<DropMessage> retrieveDropMessages(URI uri, long sinceDate) {
+        HTTPResult<Collection<byte[]>> cipherMessages = getDropMessages(uri, sinceDate);
+        Collection<DropMessage> plainMessages = new ArrayList<>();
+
+        List<Contact> ccc = new ArrayList<>();
+        for (Contacts contacts: getAllContacts().values()) {
+            ccc.addAll(contacts.getContacts());
+        }
+        Collections.shuffle(ccc, new SecureRandom());
```

Comment:
I don't think so. This implementation is pretty old and I only moved it. I'll check on that.